psyneulink/core/components/functions/transferfunctions.py

```diff
@@ -130,7 +130,40 @@ def _gen_llvm_function_body(self, ctx, builder, params, state, arg_in, arg_out,
 
         return builder
 
+    @handle_external_context()
+    def function(self,
+                 variable=None,
+                 context=None,
+                 params=None,
+                 target_set=None,
+                 derivative=False,
+                 **kwargs):
+        # Validate variable and assign to variable, and validate params
+        variable = self._check_args(variable=variable,
+                                    context=context,
+                                    params=params,
+                                    target_set=target_set,
+                                    )
+        def _get_param(p):
+            try:
+                return self._get_current_parameter_value(p.name, context)
+            except:
+                print(e)
+        modulable_params = {p.name: _get_param(p) for p in self.parameters}
+        if params is None:
+            params = {}
+        parameters = {**modulable_params, **params}
+        parameters["variable"] = variable
+        if derivative:
+            return self._derivative(**kwargs, **parameters)
+        else:
+            value = self._function(**kwargs, **parameters)
```

Comment:
This is a rather ugly extra redirection. why can't you just move the checks to a helper method and call that from both `function` and `derivative`?